internal/service/order/task/actualize.go

```diff
@@ -0,0 +1,139 @@
+package task
+
+import (
+	"context"
+	"log/slog"
+	"time"
+
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/ktigay/loyalty/internal/accrual"
+	"github.com/ktigay/loyalty/internal/db"
+	"github.com/ktigay/loyalty/internal/entity"
+	"github.com/ktigay/loyalty/internal/repository/balance"
+	"github.com/ktigay/loyalty/internal/repository/order"
+	ordersv "github.com/ktigay/loyalty/internal/service/order"
+)
+
+//go:generate mockgen -destination=./mocks/mock_order.go -package=mocks github.com/ktigay/loyalty/internal/service/order/task OrderRepoInterface
+type OrderRepoInterface interface {
+	OrdersByStatus(ctx context.Context, st ...entity.OrderStatus) (*[]entity.Order, error)
+	UpdateAll(ctx context.Context, orders []entity.Order) (*[]entity.Order, error)
+}
+
+//go:generate mockgen -destination=./mocks/mock_balance.go -package=mocks github.com/ktigay/loyalty/internal/service/order/task BalanceRepoInterface
+type BalanceRepoInterface interface {
+	IncreaseCurrent(ctx context.Context, userUUID string, delta int64) (*entity.Balance, error)
+}
+
+//go:generate mockgen -destination=./mocks/mock_status.go -package=mocks github.com/ktigay/loyalty/internal/service/order/task StatusGetterInterface
+type StatusGetterInterface interface {
+	ReceiveStatus(ctx context.Context, orders []entity.Order) (*[]entity.Order, error)
+}
+
+// ActualizeOrderTask Структура для обновления статуса заказов.
+type ActualizeOrderTask struct {
+	statusGetter StatusGetterInterface
+	orderRepo    OrderRepoInterface
+	balanceRepo  BalanceRepoInterface
+	pgxTx        db.TxFacadeInterface
+	logger       *slog.Logger
+	interval     time.Duration
+}
+
+// ActualizeOrdersStatus Актуализация данных не обработанных заказов.
+func (o *ActualizeOrderTask) ActualizeOrdersStatus(ctx, exitCtx context.Context) {
+	t := time.NewTicker(o.interval)
+	defer t.Stop()
+
+	for {
+		select {
+		case <-exitCtx.Done():
+			t.Stop()
+			o.logger.Info("Exiting ActualizeOrdersStatus")
+			return
+		case <-t.C:
+			o.actualize(ctx)
+		}
+	}
+}
+
+func (o *ActualizeOrderTask) actualize(ctx context.Context) {
+	var (
+		ordersPtr   *[]entity.Order
+		receivedPtr *[]entity.Order
+		updatedPtr  *[]entity.Order
+		cntOrders   int
+		cntReceived int
+		cntUpdated  int
+		err         error
+	)
+
+	statusesForUpdate := []entity.OrderStatus{entity.NEW, entity.PROCESSING}
+
+	o.logger.Debug("Actualizing order status")
+	ordersPtr, err = o.orderRepo.OrdersByStatus(ctx, statusesForUpdate...)
```

Comment:
Здесь мне кажется проще передать параметры по одному - в этом и есть суть variadic arguments.
```suggestion
	o.logger.Debug("Actualizing order status")
	ordersPtr, err = o.orderRepo.OrdersByStatus(ctx, entity.NEW, entity.PROCESSING)
```